Added a JFXPopup every time the start_date is greater than the end_date

AlecsisDuarte opened the pull request:
I added a little `JFXPopup` to be shown on top of the `end_date` `JFXDatePicker` control every time the user selects an end_date **lower** than the start_date as requested in the issue #3.

AlecsisDuarte committed:
Added a JFXPopup every time the start_date is greater tthan the end_date

atiqueahmedziad commented on src/App/ProjectDetail/ProjectDetailController.java:
```diff
@@ -184,6 +184,18 @@ private void onreleaseID(KeyEvent keyEvent) {
     private void showDays(ActionEvent event) throws IOException {
         getEsti_time();
```
```getEsti_time()``` is used to display number of days between Start Date and End Date in  ```JFXTextField esti_time```
We should call this method only if the end date is valid i.e. equal or greater than start date.

atiqueahmedziad commented on src/App/ProjectDetail/ProjectDetailController.java:
```diff
@@ -184,6 +184,18 @@ private void onreleaseID(KeyEvent keyEvent) {
+                Label label = new Label("End date must be equal or greater than start date");
+                JFXPopup popup = new JFXPopup(label);
+                popup.show(end_date, JFXPopup.PopupVPosition.BOTTOM, JFXPopup.PopupHPosition.RIGHT);
+                end_date.setValue(null);
```
I think you should not set ```end_date``` to ```null``` since in the next line you have placed the ```return``` keyword which eventually checking the above if condition again and ``` Date endDate``` is being assigned to ```null```, so that gives a null pointer exception error.

AlecsisDuarte replied:
What should I show if the user hasn't selected a start date, yet he chooses an end date?
The same `End date must be equal or greater than start date` label?

atiqueahmedziad replied:
Yes, I think we can use this statement (```End date must be equal or greater than start date```). Also, if we want we can use another statement i.e. ```Start date cannot be empty``` for this case only. I leave it to your decision :)

atiqueahmedziad commented on src/App/ProjectDetail/ProjectDetailController.java:
```diff
@@ -184,6 +184,18 @@ private void onreleaseID(KeyEvent keyEvent) {
+            Date endDate = Date.valueOf(end_date.getValue());
+            if (endDate.compareTo(startDate) < 0) {
+                Label label = new Label("End date must be equal or greater than start date");
+                JFXPopup popup = new JFXPopup(label);
```
The popup message is blocking the End Date JFXDatePicker which is not good to look at. I would rather suggest you to place a label field below the end date field in [Projectdetail FXML file](https://github.com/atiqueahmedziad/Easy-Project-Management-Tool/blob/master/src/App/ProjectDetail/projectdetail.fxml)
And once you get the difference between start date and end date less than 0, you can set the label text to ```End date must be equal or greater than start date```

<img width="741" alt="screenshot 2018-10-30 at 7 39 40 pm" src="https://user-images.githubusercontent.com/8364578/47721992-94c7a480-dc7b-11e8-8da9-24e429ff34c5.png">

AlecsisDuarte replied:
Do you think it should have any color to call the **users attention**?

atiqueahmedziad replied:
@AlecsisDuarte I think we can use red color text to grab attention 😃 .

atiqueahmedziad reviewed (changes requested):
Thanks for the pull request. I have added some note. Let me know when you make the require changes in the pull request. 😄 
We are also selecting start date and end date for a task of a project from [AddTaskDetail](https://github.com/atiqueahmedziad/Easy-Project-Management-Tool/blob/master/src/App/AddTask/AddtaskController.java#L120) method. Can you also validate the end date there before saving the task info in the database ?

AlecsisDuarte committed:
Added the same validation to AddTaskController and beautified the error popup/label

AlecsisDuarte commented:
I made a few many changes:
- Added the dates validation to the `AddTask` method as requested in https://github.com/atiqueahmedziad/Easy-Project-Management-Tool/pull/6#discussion_r229304912
- Changed from a `JFXPopup` to a Label in both controllers and changed the color to **red** :red_circle: 
- Implemented a date validation method that returns whether the date range is :heavy_check_mark:  or :x: 
- And found a way to avoid getting null pointer exceptions by always validating it isn't as noted in https://github.com/atiqueahmedziad/Easy-Project-Management-Tool/pull/6#discussion_r229307765

Hope this fulfills the requirements and if not just let me know :smile:

atiqueahmedziad reviewed (approved):
All good 😄 .

atiqueahmedziad commented:
Thanks a lot for your contribution @AlecsisDuarte . 😀

AlecsisDuarte commented:
You are welcome @atiqueahmedziad, great I helped
